Ruins/src/main/java/atomicstryker/ruins/common/RuinTemplate.java

```diff
@@ -465,26 +470,30 @@ else if (curRule.runLast())
 
         for (AdjoiningTemplateData ad : adjoiningTemplates)
         {
-            debugPrinter.printf("Considering to spawn adjoining %s of Ruin %s...\n", ad.adjoiningTemplate.getName(), getName());
+        	RuinTemplate adTemplate = ad.adjoiningTemplates.get((int)(world.rand.nextFloat() * ad.adjoiningTemplates.size()));
+        	
+            debugPrinter.printf("Considering to spawn adjoining %s of Ruin %s...\n", adTemplate.getName(), getName());
             float randres = (world.rand.nextFloat() * 100);
             if (randres < ad.spawnchance)
             {
                 int newrot = world.rand.nextInt(4);
                 int targetX = xBase + ad.relativeX;
                 int targetZ = zBase + ad.relativeZ;
-                int targetY = ad.adjoiningTemplate.checkArea(world, targetX, y, targetZ, newrot, ad.acceptableY);
-                if (targetY > 0 && Math.abs(y - targetY) <= ad.acceptableY)
+                int targetY;
+                if (ad.useRelativeY) targetY = y + ad.relativeY;

```

Comment:
Minor point, but you break the formatting rules here (i have a formatter in my workspace, no? Yes i do)